### ISSUE TITLE
using /ASK still writing code 

### ISSUE THREAD

**Issue by Yona544** [paul-gauthier/aider#1303]
### Issue

I had it more than once using /ASK in the browser and still, aider made changes to the code 
In the terminal I can see this error 

litellm.InternalServerError: AnthropicException - Overloaded. Handle with `litellm.InternalServerError`.
Retrying in 0.2 seconds...

not sure if it is related, maybe after the error it forgets that we are in /ASK 

### Version and model info

Aider v0.54.12
Main model: claude-3-5-sonnet-20240620 with diff edit format, infinite output
Weak model: claude-3-haiku-20240307
Git repo: \192.168.0.35\YonaVmDataShare\Projects\KDSMenu.git with 51 files
Repo-map: using 1024 tokens, auto refresh

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

The `/ask` command is not able to edit code. But keep in mind `/ask` only applies to the one question you include after it. If you type another message after, you're back in code mode. Use `/chat-mode ask` to switch to ask mode.

`litellm.InternalServerError` means Anthropic's servers are down/overloaded. Unrelated to the ask issue.

**Comment by Yona544** [comment]
I'm aware that /ask only applies to the one question. I had an issue with one question only. 
I suspect that after some error, regardless of the source aider resends the request but forgets that we are in /ASK mode

**Comment by fry69** [comment]
> I'm aware that /ask only applies to the one question. I had an issue with one question only. I suspect that after some error, regardless of the source aider resends the request but forgets that we are in /ASK mode

I asked aider to look at itself (using Sonnet-3.5) and it said the edit format (`/ask`) gets preserved through retries ->

---

After carefully examining the code, I can conclude that the `/ask` context is preserved during retries. Here's why:

1. When the `/ask` command is used, a new `Coder` instance is created with the "ask" edit format.
2. The `send_message` method adds the user's input to `cur_messages` and then formats all messages, including the edit format context.
3. If a LiteLLM temporary error occurs, the code enters a retry loop. This loop only retries the `self.send(messages, functions=self.functions)` call, which uses the same `messages` that were initially formatted.
4. The retry loop does not modify the `Coder` instance or its edit format. It simply waits and tries to send the same messages again.

Therefore, even if there are multiple retries due to temporary errors, the `/ask` context (i.e., the "ask" edit format) is preserved throughout the process. The question will still be sent in `/ask` mode, not in edit mode.

The bug report suggesting that the question was sent in edit mode after a retry is likely due to a different issue or misunderstanding. Based on this code, it's not possible for the edit format to change from "ask" to edit mode during retries.

To summarize: The `/ask` context is preserved during retries, and it's not possible for the question to be sent in edit mode after a retry if it was initially sent in `/ask` mode.

Tokens: 90k sent, 1.2k received. Cost: $0.29 message, $0.29 session.

---

**Comment by Yona544** [comment]
Yes, the workflow of asking for updates and then implementing them has been incredibly powerful for me. I've been using it extensively.

However, I've encountered more than once where the actual code gets updated unexpectedly. I'm using the browser, so I’m wondering if that could be contributing to the problem.

Additionally, I've noticed that when using the terminal and pressing **/,** a list of all available options is displayed. It would be incredibly useful if something similar could be added to the browser, or perhaps even a **/chat-mode** checkbox.

Thanks for your help and for this wonderful project!

**Comment by paul-gauthier** [comment]
I will keep an eye out for this problem, but I am currently unable to reproduce it. If you can describe steps to make it happen, please let me know.

I'm going to close this issue for now, but feel free to add a comment here and I will re-open or file a new issue any time.

**Comment by Yona544** [comment]
Just had it once more. Will send you the pictures
![aid1](https://github.com/user-attachments/assets/9c85a002-b93a-422c-8592-312594e672aa)
![aid2](https://github.com/user-attachments/assets/c3dd5a05-5c51-4e12-8ad3-111b0db9e879)
![aid3](https://github.com/user-attachments/assets/83c0dcb3-caa3-4e1e-b663-bdb4f6f9922a)
![aid4](https://github.com/user-attachments/assets/29e1b846-3704-459e-af6e-1566d08c95d4)

**Comment by paul-gauthier** [comment]
Unfortunately, none of the `/` commands are currently available in the web browser version.

**Comment by Yona544** [comment]
Any plan to add it?

**Comment by paul-gauthier** [comment]
Yes, I hope to!
